src/main/appender.cpp

```diff
@@ -375,6 +411,69 @@ void Appender::FlushInternal(ColumnDataCollection &collection) {
 	context->Append(*description, collection);
 }
 
+void Appender::AppendDefaultsToVector(Vector &result, idx_t column, SelectionVector &sel, idx_t count) {
+#ifdef DEBUG
+	for (idx_t i = 0; i < count; i++) {
+		auto index = sel.get_index(i);
+		if (index > STANDARD_VECTOR_SIZE) {
+			// FIXME: should we require the size of the 'result' Vector as well?
+			throw InternalException("Provided SelectionVector is invalid, index %d points to %d, which is out of range",
+			                        i, index);
+		}
+	}
```

Comment:
Maybe we can move this into a `Verify` function on the selection vector itself? We can also use it in other places where we construct selection vectors. The vector size might already be sufficiently tested by the `vector.Verify(count)` functions?

Reply:
I agree, I think the FIXME should also be applied, we need to know the size of the vector it's being applied to if we want to do meaningful error checking